src/dotnet-core-uninstall/Windows/RegistryQuery.cs

```diff
@@ -40,65 +42,81 @@ public static IEnumerable<Bundle> GetAllInstalledBundles()
                 && IsDotNetCoreBundleUninstaller(registryKey.GetValue("WindowsInstaller") as int?)
                 && IsNotVisualStudioDummyVersion(registryKey.GetValue("DisplayName") as string);
         }
 
-        private static bool IsNotVisualStudioDummyVersion(string displayName)
+        private bool IsNotVisualStudioDummyVersion(string displayName)
         {
             return !displayName.Contains(" from Visual Studio");
         }
 
-        private static bool IsDotNetCoreBundleDisplayName(string displayName)
+        private bool IsDotNetCoreBundleDisplayName(string displayName)
         {
             return displayName == null ?
                 false :
                 Regexes.BundleDisplayNameRegex.IsMatch(displayName);
         }
 
-        private static bool IsDotNetCoreBundlePublisher(string publisher)
+        private bool IsDotNetCoreBundlePublisher(string publisher)
         {
             return publisher == null ?
                 false :
                 Regexes.BundlePublisherRegex.IsMatch(publisher);
         }
 
-        private static bool IsDotNetCoreBundleUninstaller(int? windowsInstaller)
+        private bool IsDotNetCoreBundleUninstaller(int? windowsInstaller)
         {
             return windowsInstaller == null;
         }
 
-        private static Bundle WrapRegistryKey(RegistryKey registryKey)
+        private Bundle WrapRegistryKey(RegistryKey registryKey)
         {
             var displayName = registryKey.GetValue("DisplayName") as string;
             var uninstallCommand = registryKey.GetValue("QuietUninstallString") as string;
             var bundleCachePath = registryKey.GetValue("BundleCachePath") as string;
 
-            ParseVersionAndArch(displayName, bundleCachePath, out var version, out var arch);
+            ParseVersionAndArch(registryKey, displayName, bundleCachePath, out var version, out var arch);
+
+            if (version == null)
+            {
+                return null;
+            }
 
             return Bundle.From(version, arch, uninstallCommand, displayName);
         }
 
-        private static void ParseVersionAndArch(string displayName, string bundleCachePath, out BundleVersion version, out BundleArch arch)
+        private void ParseVersionAndArch(RegistryKey registryKey, string displayName, string bundleCachePath, out BundleVersion version, out BundleArch arch)
         {
             var match = Regexes.BundleDisplayNameRegex.Match(displayName);
             var cachePathMatch = Regexes.BundleCachePathRegex.Match(bundleCachePath);
             var archString = cachePathMatch.Groups[Regexes.ArchGroupName].Value ?? string.Empty;
-            var versionString = cachePathMatch.Groups[Regexes.VersionGroupName].Value;
+            var versionFromCachePath = cachePathMatch.Groups[Regexes.VersionGroupName].Value;
+            var versionFromRegistry = string.Join('.', (registryKey.GetValue("DisplayVersion") as string).Split('.').Take(3));
```

Comment:
could you help me understand this logic? Maybe add some comment, why we can also get the result from CachePath

Reply:
Sure, this is essentially the underlying problem with #103 and #106. The cache path for asp.net runtimes doesn't have the version in it so the version can't be parsed out in the same way that it is for the other bundle types. This was causing an error when you tried to `list` on a machine with these runtimes, since it couldn't be parsed in the expected way. 
I'm fixing this here by getting the version from the registry, since this is a more reliable way to get the version for asp.net runtimes.
Are you suggesting adding a comment in the code about this?

Reply:
Yes, it is a good place to add comment. It is an external system's behavior and it is not well known at all

Reply:
btw there is a way to "document" by test, but it is over kill in this case

Blog post from a friend of mine
https://docs.microsoft.com/en-us/archive/blogs/ericgu/portadaptersimulator-read-only-and-write-only-dependencies